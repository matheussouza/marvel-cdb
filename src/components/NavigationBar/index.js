import React from 'react';

const NavigationBar = ({ page, total, search, changePage, pageSize = 20 }) => {
  return (
    <div style={{ display: 'flex', justifyContent: 'center', width: '100%', paddingBottom: 20 }}>
      <div style={{ display: 'flex' }}>
        <button
          disabled={ page === 1 } 
          className="btn btn-danger"
          style={{ marginRight: 20, width: 56 }}
          onClick={() => changePage(search, page - 1)}
        >
          <i className="fa fa-chevron-left" />
        </button>
        <button
          disabled={ total <= pageSize || page === Math.ceil(total / pageSize) }
          className="btn btn-danger"
          style={{ marginLeft: 20, width: 56 }}
          onClick={() => changePage(search, page + 1)}
        >
          <i className="fa fa-chevron-right" />
        </button>
      </div>
    </div>
  );
};

export default NavigationBar;