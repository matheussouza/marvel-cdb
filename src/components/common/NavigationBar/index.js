import React from 'react';

export default ({ page, total, search, changePage, pageSize = 20 }) => {
  if (!page || page === 0 )
    return <></>;

  return (
    <div className="d-flex app-navigation-bar-container" >
      <div className="d-flex">
        <button
          data-testid="previous-page-button"
          disabled={ page === 1 } 
          className="btn btn-danger app-navigation-button"
          onClick={() => changePage(search, page - 1)}
        >
          <i className="fa fa-chevron-left" />
        </button>
        <button
          data-testid="next-page-button"
          disabled={ total <= pageSize || page === Math.ceil(total / pageSize) }
          className="btn btn-danger app-navigation-button"
          onClick={() => changePage(search, page + 1)}
        >
          <i className="fa fa-chevron-right" />
        </button>
      </div>
    </div>
  );
};
