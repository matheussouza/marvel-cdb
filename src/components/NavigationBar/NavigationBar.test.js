import React from 'react';
import { render } from '@testing-library/react';

import NavigationBar from './index';

describe('NavigationBar component', () => {

  let mockChangePage;

  beforeEach(() => {
    mockChangePage = jest.fn();
  });

  it('should be empty component with no arguments', () => {
    const subject = render(<NavigationBar />);
    expect(subject.container.innerHTML).toMatch("");
  });

  it('should be empty component in page 0', () => {
    const subject = render(<NavigationBar page={0} />);
    expect(subject.container.innerHTML).toMatch("");
  });

  it('should be equals snapshot', () => {
    const subject = render(<NavigationBar page={1} total={10} search="" changePage={mockChangePage} />);
    expect(subject.container).toMatchSnapshot();
  });

  it('should have disabled previous button in first page', () => {
    const subject = render(<NavigationBar page={1} />);

    expect(subject.getByTestId('previous-page-button')).toHaveProperty('disabled', true);
  });

  it('should have enabled previous button in second page', () => {
    const subject = render(<NavigationBar page={2} />);

    expect(subject.getByTestId('previous-page-button')).toHaveProperty('disabled', false);
  });

  it('should have disabled next button with total less than page size', () => {
    const subject = render(<NavigationBar page={1} total={10} />);

    expect(subject.getByTestId('next-page-button')).toHaveProperty('disabled', true);
  });

  it('should have disabled next button with total equals page size', () => {
    const subject = render(<NavigationBar page={1} total={10} pageSize={10} />);

    expect(subject.getByTestId('next-page-button')).toHaveProperty('disabled', true);
  });

  it('should have enabled next button when total greater than page size', () => {
    const subject = render(<NavigationBar page={2} total={40} pageSize={10} />);

    expect(subject.getByTestId('next-page-button')).toHaveProperty('disabled', false);
  });

  it('should have disabled next button when in last page', () => {
    const subject = render(<NavigationBar page={2} total={40} />);

    expect(subject.getByTestId('next-page-button')).toHaveProperty('disabled', true);
  });

});