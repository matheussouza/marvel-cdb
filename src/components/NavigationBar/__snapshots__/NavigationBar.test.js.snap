// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component should be equals snapshot 1`] = `
<div>
  <div
    class="d-flex app-navigation-bar-container"
  >
    <div
      class="d-flex"
    >
      <button
        class="btn btn-danger app-navigation-button"
        data-testid="previous-page-button"
        disabled=""
      >
        <i
          class="fa fa-chevron-left"
        />
      </button>
      <button
        class="btn btn-danger app-navigation-button"
        data-testid="next-page-button"
        disabled=""
      >
        <i
          class="fa fa-chevron-right"
        />
      </button>
    </div>
  </div>
</div>
`;
